internal/core/core.go

```diff
@@ -570,38 +584,33 @@ func (v VDSHandle) GetAttributes(
 		cAttributes[i] = C.enum_attribute(targetAttributes[i])
 	}
 
-	cPrimarySurface, err := primarySurface.ToCRegularSurface(nil)
+	cPrimarySurfaceData, err := primarySurface.toCdata(0)
```

Comment:
What is the benefit of having both `toCdata` and `toCRegularSurface` over just

`func newCRegularSurface(surface RegularSurface, shift float) (CRegularSurface, error)`

(If it's because it gets big, you can split out the C array stuff in its own helper function that's used internally)

To make caller code as nice as possible I would preferably have an overload like this too:

```
func newCRegularSurface(surface RegularSurface) (CRegularSurface, error) {
    return newCRegularSurface(surface, 0)
}
```

Reply:
Gad damn golang, why you no have overloads?